app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -0,0 +1,161 @@
+    private lateinit var progressBar: View
+    private var subscription: Disposable? = null
+
+    init {
+        val activity = unwrapToActivityContext(getContext())
+        val component = obtain(activity)
+        service = component.service()
+        navigate = component.navigator()
+        analytics = component.analytics()
+
+        viewPagerAdapter = ScheduleViewPagerAdapter(activity)
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        progressBar = findViewById(R.id.progressbar)
+
+        val viewPager = findViewById<ViewPager>(R.id.viewpager)
+        val tabLayout = findViewById<TabLayout>(R.id.tabstrip)
+        tabLayout.setupWithViewPager(viewPager)
+        hackToApplyTypefaces(tabLayout)
+
+        viewPager.adapter = viewPagerAdapter
+
+        tabLayout.addOnTabSelectedListener(TrackingOnTabSelectedListener(analytics, viewPagerAdapter))
+
+        setupToolbar()
+    }
+
+    private fun setupToolbar() {
+        val toolbar = findViewById<Toolbar>(R.id.toolbar)
+        toolbar.setTitle(R.string.activity_schedule)
+        toolbar.inflateMenu(R.menu.homepage)
+        toolbar.setOnMenuItemClickListener { item ->
+            when (item.itemId) {
+                R.id.action_search -> {
+                    navigate.toSearch()
+                    return@setOnMenuItemClickListener true
+                }
+                R.id.action_settings -> {
+                    navigate.toSettings()
+                    return@setOnMenuItemClickListener true
+                }
+                else -> return@setOnMenuItemClickListener false
+            }
+        }
+    }
+
+    override fun startLoading() {
+        subscription = service.schedule(false)
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(
+                    {
+                        updateWith(
+                                it,
+                                object : ScheduleViewPagerAdapter.OnEventClickedListener {
+                                    override fun onEventClicked(event: Event) = onEventClickeds(event)
+                                }
+                        )
+                    },
+                    { Timber.e(it) }
+            )
+    }
+
+    private fun onEventClickeds(event: Event) {
+        analytics.trackItemSelected(ContentType.SCHEDULE_ITEM, event.id)
+        navigate.toEventDetails(event.id)
+    }
+
+    override fun stopLoading() {
+        subscription!!.dispose()
+    }
+
+    private fun hackToApplyTypefaces(tabLayout: TabLayout) {
+        // Unfortunately doing this the sensible way (in ScheduleViewPagerAdapter.getPageTitle())
+        // results in a bunch of other views on screen to stop drawing their text, for reasons only
+        // known to the Gods of Kobol. We can't theme things in the TabLayout either, as the
+        // TextAppearance is applied _after_ inflating the tab views, which means Calligraphy can't
```

Comment:
This is old stuff that mentions Calligraphy, would you mind checking if it's still needed now that we've ditched it, and if it is, to replace Calligraphy with Fonts Support Library?

Reply:
I'll add an issue for that, I don't want to address it here as this PR has been opened enough

Reply:
https://github.com/squanchy-dev/squanchy-android/issues/405